plugin/core/types.py

```diff
@@ -296,6 +300,39 @@ def document_highlight_style_region_flags(self) -> Tuple[int, int]:
         else:
             return sublime.DRAW_NO_FILL, sublime.DRAW_NO_FILL | sublime.DRAW_NO_OUTLINE | sublime.DRAW_SOLID_UNDERLINE
 
+    @staticmethod
+    def _style_str_to_flag(style_str: str) -> int:
+        # This method could be a dict or lru_cache
+        if style_str == "":
+            return sublime.DRAW_NO_FILL | sublime.DRAW_NO_OUTLINE
+        elif style_str == "box":
+            return sublime.DRAW_NO_FILL
+        elif style_str == "underline":
+            return sublime.DRAW_NO_FILL | sublime.DRAW_NO_OUTLINE | sublime.DRAW_SOLID_UNDERLINE
+        elif style_str == "stippled":
+            return sublime.DRAW_NO_FILL | sublime.DRAW_NO_OUTLINE | sublime.DRAW_STIPPLED_UNDERLINE
+        elif style_str == "squiggly":
+            return sublime.DRAW_NO_FILL | sublime.DRAW_NO_OUTLINE | sublime.DRAW_SQUIGGLY_UNDERLINE
+        else:
+            return 0  # default styling
+
+    def diagnostics_highlight_style_flags(self) -> List[int]:
+        """Returns flags for highlighting diagnostics on single lines per severity"""
+        if isinstance(self.diagnostics_highlight_style, str):
+            # same style for all severity levels
+            return [self._style_str_to_flag(self.diagnostics_highlight_style)] * 4
+        elif isinstance(self.diagnostics_highlight_style, dict):
+            flags = []
+            for sev in ("error", "warning", "info", "hint"):
+                user_style = self.diagnostics_highlight_style.get(sev)
+                if user_style is None:  # user did not provide a style
+                    flags.append(0)
```

Comment:
How is the flag `0` rendered? Is `0` an appropriate default?

Reply:
I guess it is used because it evaluates to "false", and that allows to use a convenient way to get the actual value with an `or` or `if ... else` in the other function below. But maybe it would be more readable to do this logic right here and just append the actual default value for each severity instead of `0`.

Reply:
@jwortmann You are correct. The scheme takes advantage of the short-circuiting behaviour of python's `or` operator to make the code less verbose. 

Regarding moving the logic here, I considered that, but decided against it. The reason being, this class/file is only supposed to model _user settings_, at least from what I understood. If we start adding custom logic here, it might open the floodgates to this class being a dumping ground for more logic.